tools/clang/unittests/HLSL/ValidationTest.cpp

```diff
@@ -682,9 +682,9 @@ TEST_F(ValidationTest, DeadLoopFail) {
   RewriteAssemblyCheckMsg(
       L"..\\CodeGenHLSL\\loop1.hlsl", "ps_6_0",
       {"br i1 %exitcond, label %for.end.loopexit, label %for.body, !llvm.loop !([0-9]+)",
-       "%add.lcssa = phi float \\[ %add, %for.body \\]",
+       "?%add[.lcssa]? = phi float \\[ %add, %for.body \\]",
        "!dx.entryPoints = !\\{!([0-9]+)\\}",
-       "\\[ %add.lcssa, %for.end.loopexit \\]"
+       "\\[ %add[.lcssa]?, %for.end.loopexit \\]"
```

Comment:
I don't think `[.lcssa]?` is the right syntax for what you want here - this says it expects zero or one characters from this set.  I think you want `(.lcssa)?`.